Add TRE Open OnDemand Service documentation #217

mikej888 opened the pull request:
# EIDF Documentation Pull Request

## Description

Added documentation for TRE Open OnDemand Service to new folders:

* `docs/safe-haven-services/open-ondemand/`, documentation
* `docs/images/open-ondemand/`, images

Updated  `mkdocs.yml`, menu side-bar with links to access documentation under a 'Safe Haven Services' => 'TRE Container User Guide' => 'TRE Open OnDemand service' menu.

Fixes #217

## Type of change

- [x] New Documentation

## What has to be reviewed

All content under `docs/safe-haven-services/open-ondemand/`. Please add comments and suggestions to this pull request.

Note: As OOD requires users to be familiar with job scheduling (which is a given for OOD's intended users but not for ours!) I've dealt with this in as simple and minimal way as possible (for now)

Note: ``docs/safe-haven-services/open-ondemand/containers.md` currently has text

`Open OnDemand uses the Container Execution Service tools to run containers. Consequently, containers run via Open OnDemand **must** conform to the requirements of the Container Execution Service. See the [TRE Container User Guide](../tre-container-user-guide/introduction.md).`

When @giuliadei's updated CES docs have been merged into EPCCed/eidf-docs,  this link may need to change accordingly.

## Checklist

- [x] Documentation follows the project style guidelines
- [x] Self-review of documentation using mkdocs on local system
- [x] Spellcheck has been performed
- [x] Pre-commit has been run and passed (pre-commit has 2 Markdown failures caused by others' documentation elsewhere, not my content, see #221)

rkm commented on docs/safe-haven-services/open-ondemand/containers.md:
```diff
@@ -0,0 +1,109 @@
+
+    `/safe_data/PROJECT_SUBDIRECTORY` is **not** available on TRE-level back-ends (e.g., the Superdome Flex). For these, you will need to stage your data to the TRE-level back-end following your project- and safe haven-specific processes for the use of TRE-level services.
+
+### Troubleshooting: 'Cannot open project data: /safe_data/cannot_determine_project_from_groups'
```
We should add that the user must contact their RC if they encounter this error.

mikej888 replied:
Updated `open-ondemand/containers.md` to direct users to their Research Coordinator if a project-specific `/safe_data/` folder can't be identified.

mikej888 commented:
> General feedback:
>
> * There might be overlap in some areas with OOD's own documentation. Might it be easier to link to these external docs and note the specifics of our deployment?

Open OnDemand has a product site, https://openondemand.org/, and installation/configuration docs, https://osc.github.io/ood-documentation/, but no user docs, I'd assume because its branding and apps is very site-specific. Sites just roll their own e.g.,

* https://hpcdocs.hpc.arizona.edu/running_jobs/open_on_demand/
* https://www.win.ox.ac.uk/research/it/i-want-to/wfh/remote-desktops/fmrib-remote-desktops/fmrib-ondemand
* https://docs.pages.surrey.ac.uk/research_computing/hpc/ood/
* https://www.rc.virginia.edu/userinfo/hpc/ood/
* https://docs.rcc.fsu.edu/hpc/ood/

* I did look through these for 'inspiration'.

mikej888 commented:
Additional changes applied:

* Fixed typo in open-ondemand/containers.md: DockerHub => GitHub
* Fixed typo in open-ondemand/jobs.md: 'acess' => 'access'
* Added ECDF GitLab to open-ondemand/containers.md 'Container registries' table
* Updated open-ondemand/access.md: 'research coordinator' => 'Research Coordinator'

Addressing other comments still to be done.

mikej888 commented on docs/safe-haven-services/open-ondemand/access.md:
```diff
@@ -0,0 +1,39 @@
+    * Click **Yes**
+1. If using Firefox and a Warning: Potential Security Risk Ahead page appears with text 'Error code: SEC_ERROR_UNKNOWN_ISSUER', then
+    1. Click **Advanced...**
+    1. Click **Accept the Risk and Continue**
```
I'd hope that these warnings will cease to appear at some point! In the meantime, I've updated `open-ondemand/access.md` with an explanation as to why ignoring OOD certificate warnings is OK in this case.

mikej888 commented on docs/safe-haven-services/open-ondemand/jobs.md:
```diff
@@ -0,0 +1,335 @@
+
+---
+
+## Back-ends, clusters, jobs and apps
```
Skipping for now until its clearer what back-ends OOD deployments will have access to.

mikej888 commented:
@rkm wrote
> General feedback:
> 
> * Some pages e.g., Jobs contain quite a depth of technical detail, which is useful but perhaps not the best format for a new user . It might be beneficial to have a high-level introduction with a basic job submission workflow as the initial page.

Great suggestion, thanks! I've added `open-ondemand/getting-started.md` with a tutorial introduction to the basics of OOD and jobs centred round running both the batch container app and and JupyterLab app. If you could give that a quick look that would be great, thanks!
